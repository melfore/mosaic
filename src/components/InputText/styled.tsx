import { TextField as MUITextField, styled } from "@material-ui/core";

export const StyledMUITextField = styled(MUITextField)({
  width: "100%",
});
