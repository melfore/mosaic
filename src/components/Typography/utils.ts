type VariantDefaultsType = {
  [key: string]: {
    bottomSpacing: boolean;
    truncated: boolean;
  };
};

const VARIANT_DEFAULTS: VariantDefaultsType = {
  body1: {
    bottomSpacing: false,
    truncated: false,
  },
  caption: {
    bottomSpacing: false,
    truncated: false,
  },
  h5: {
    bottomSpacing: true,
    truncated: true,
  },
  h6: {
    bottomSpacing: true,
    truncated: true,
  },
  overline: {
    bottomSpacing: false,
    truncated: false,
  },
  subtitle1: {
    bottomSpacing: false,
    truncated: false,
  },
};

export const getBottomSpacing = (variant: string, bottomSpacing?: boolean): boolean =>
  bottomSpacing === undefined ? VARIANT_DEFAULTS[variant].bottomSpacing : bottomSpacing;

export const getTruncate = (variant: string, truncated?: boolean): boolean =>
  truncated === undefined ? VARIANT_DEFAULTS[variant].truncated : truncated;

export const VARIANT_COMPONENT_MAP = {
  body1: "p",
  caption: "span",
  h5: "h1",
  h6: "h2",
  overline: "h4",
  subtitle1: "h3",
};
