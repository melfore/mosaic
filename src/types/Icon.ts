import { LoadableType } from "./Base";

export enum Icons {
  account = "account",
  add = "add",
  apps = "apps",
  block = "block",
  brightness = "brightness",
  business = "business",
  chart_timeline = "chart_timeline",
  check = "check",
  checkbox = "checkbox",
  checkbox_empty = "checkbox_empty",
  clock = "clock",
  close = "close",
  delete = "delete",
  down = "down",
  edit = "edit",
  error = "error",
  filter = "filter",
  first = "first",
  home = "home",
  last = "last",
  left = "left",
  list = "list",
  list_check = "list_check",
  mail = "mail",
  mail_open = "mail_open",
  menu = "menu",
  notifications = "notifications",
  open_new = "open_new",
  pause = "pause",
  play = "play",
  print = "print",
  refresh = "refresh",
  right = "right",
  save = "save",
  search = "search",
  send = "send",
  settings = "settings",
  up = "up",
  up_arrow = "up_arrow",
  warn = "warn",
}

export enum IconSize {
  default = "default",
  large = "large",
  small = "small",
}

interface IconForwardedType {
  className?: string;
  color?: string;
  ref?: any;
}

export interface IconType extends LoadableType {
  forwarded?: IconForwardedType;
  name: Icons;
  size?: IconSize;
}
