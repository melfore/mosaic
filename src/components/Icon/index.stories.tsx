import React from 'react';
import {  } from '@storybook/addon-actions';
import { select } from '@storybook/addon-knobs';
import { Icons, IconSize } from '../../types/Icon';
import { DOCS_PAGE_STRUCTURE, StoriesWrapper } from '../../utils/storybook';
import Icon from '.';

export default {
  title: 'Icon',
  component: Icon,
  parameters: {
    ...DOCS_PAGE_STRUCTURE,
  },
}

export const Canvas = () => (
  <Icon
    dataCy="icon"
    name={select('icon', Icons, Icons.add)}
    size={select('size', IconSize, IconSize.default)}
  />
);

export const Size = () => (
  <StoriesWrapper>
    <Icon
      dataCy="icon-send"
      name={Icons.send}
      size={IconSize.small}
    />
    <Icon
      dataCy="icon-send"
      name={Icons.send}
    />
    <Icon
      dataCy="icon-send"
      name={Icons.send}
      size={IconSize.large}
    />
  </StoriesWrapper>
);

const allIcons = Object
  .keys(Icons)
  .map((icon: string) => (
    <div className="icon-wrapper">
      <Icon
        dataCy={`icon-${icon}`}
        name={Icons[icon]}
      />
      <span>
        {icon}
      </span>
    </div>
  ));

export const AllIcons = () => (
  <StoriesWrapper>
    {allIcons}
  </StoriesWrapper>
);
