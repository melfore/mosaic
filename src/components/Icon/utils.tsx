import React from "react";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUISendIcon from "@material-ui/icons/Send";
import { Icons } from "../../types/Icon";

export const iconsCatalog = {
  [Icons.add]: <MUIAddIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.playlistAddCheck]: <MUIPlaylistAddCheckIcon />,
  [Icons.send]: <MUISendIcon />,
};
