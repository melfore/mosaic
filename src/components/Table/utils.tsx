import React, { forwardRef, Fragment } from "react";
import { Action as MTActionType, Column as MTColumnType, Options as MTOptionsType } from "material-table";
import { Icons, IconSize } from "../../types/Icon";
import { TableActionScope, TableActionType, TableColumnType } from "../../types/Table";
import Button from "../Button";
import Icon from "../Icon";
import IconButton from "../IconButton";

// For default props refer to Options section here https://material-table.com/#/docs/all-props
export const DEFAULT_TABLE_OPTIONS: MTOptionsType = {
  debounceInterval: 500,
  draggable: false,
  emptyRowsWhenPaging: false,
};

export const actionAdapter = (action: TableActionType): MTActionType<object> => {
  const {
    callback,
    label,
    disabled = false,
    hidden = false,
    icon = undefined,
    scope = TableActionScope.default,
  } = action;
  return {
    disabled,
    hidden,
    icon: `${label}$${icon}`,
    isFreeAction: scope === TableActionScope.default,
    onClick: callback,
  };
};

export const actionComponentAdapter = (props: any) => {
  const { data, action } = props;
  const { disabled, hidden, icon, isFreeAction, onClick }: MTActionType<object> = action;
  if (hidden) {
    return null;
  }

  const labelAndIconName = icon as string;
  const label = labelAndIconName.split("$")[0];
  const iconName = labelAndIconName.split("$")[1] as Icons;
  return (
    <Fragment>
      {isFreeAction && (
        <Fragment>
          <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
          <Button
            disabled={disabled}
            icon={!iconName ? undefined : { name: iconName }}
            label={label}
            onClick={(event: any) => {
              onClick(event, data);
              event.preventDefault();
              event.stopPropagation();
            }}
          />
        </Fragment>
      )}
      {!isFreeAction && (
        <Fragment>
          <IconButton
            disabled={disabled}
            icon={iconName}
            onClick={(event: any) => {
              onClick(event, data);
              event.preventDefault();
              event.stopPropagation();
            }}
            size={IconSize.small}
          />
          <span>&nbsp;&nbsp;</span>
        </Fragment>
      )}
    </Fragment>
  );
};

export const columnAdapter = (column: TableColumnType): MTColumnType<object> => {
  const { label, path, render } = column;
  return {
    field: path,
    render,
    title: label,
  };
};

export const iconAdapter = (name: Icons, size = IconSize.default) =>
  forwardRef((props: any, ref: any) => {
    return <Icon forwarded={{ ...props, ref }} name={name} size={size} />;
  });
