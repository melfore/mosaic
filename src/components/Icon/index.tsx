import React, { FC } from "react";
import { IconType, IconSize } from "../../types/Icon";
import { iconsCatalog } from "./utils";

const Icon: FC<IconType> = ({ dataCy, forwarded = {}, name, size = IconSize.default }) => {
  const icon = iconsCatalog[name];
  return React.cloneElement(icon, { ...forwarded, "data-cy": dataCy, fontSize: size });
};

export default Icon;
