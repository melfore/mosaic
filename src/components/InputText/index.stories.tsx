import React from "react";
import { action } from "@storybook/addon-actions";
import { boolean, text, object, select } from "@storybook/addon-knobs";
import { InputSize, InputVariant, InputDataType } from "../../types/Input";
import IntlProviderMock, { LocaleMock, MessageMock } from "../../utils/mocks/IntlProviderMock";
import { getDocsPageStructure, StoriesWrapper } from "../../utils/stories";
import InputText, { InputTextIntl } from ".";
import { Icons } from "../../types/Icon";

export default {
  title: "InputText",
  component: InputText,
  parameters: {
    ...getDocsPageStructure("InputText"),
  },
};

export const Canvas = () => (
  <InputText
    adornment={object("adornment", undefined)}
    dataCy="input-text"
    disabled={boolean("disabled", false)}
    initialValue={text("initialValue", "")}
    label={text("label", "Label")}
    multiline={object("multiline", undefined)}
    onChange={action("onChange callback")}
    placeholder={text("placeholder", "Placeholder Text")}
    required={boolean("required", false)}
    shrink={boolean("shrink", false)}
    size={select("size", InputSize, InputSize.default)}
    type={select("type", InputDataType, InputDataType.default)}
    variant={select("variant", InputVariant, InputVariant.default)}
  />
);

export const Disabled = () => (
  <StoriesWrapper>
    <InputText dataCy="input-text" disabled label="Label" />
    <InputText dataCy="input-text" disabled initialValue="Disabled" label="Label" />
  </StoriesWrapper>
);

export const Multiline = () => (
  <StoriesWrapper>
    <InputText
      dataCy="input-text"
      initialValue="Multiline by one"
      label="Label"
      multiline={{
        rows: 1,
      }}
    />
    <InputText
      dataCy="input-text"
      initialValue="Multiline by three"
      label="Label"
      multiline={{
        rows: 3,
      }}
    />
    <InputText
      dataCy="input-text"
      initialValue="Multiline max five, broken in MUI"
      label="Label"
      multiline={{
        rows: 3,
        rowsMax: 5,
      }}
    />
  </StoriesWrapper>
);

export const Required = () => (
  <StoriesWrapper>
    <InputText dataCy="input-text" initialValue="Required" label="Label" required />
  </StoriesWrapper>
);

export const Size = () => (
  <StoriesWrapper>
    <InputText dataCy="input-text" initialValue="Default" label="Label" />
    <InputText dataCy="input-text" initialValue="Small" label="Label" size={InputSize.small} />
  </StoriesWrapper>
);

export const Variant = () => (
  <StoriesWrapper>
    <InputText dataCy="input-text" initialValue="Default" label="Label" />
    <InputText dataCy="input-text" initialValue="Filled" label="Label" variant={InputVariant.filled} />
  </StoriesWrapper>
);

export const WithAdornment = () => (
  <StoriesWrapper>
    <InputText
      adornment={{ icon: Icons.search }}
      dataCy="input-text"
      initialValue="Adornment used to render icon"
      label="Label"
    />
    <InputText
      adornment={{ icon: Icons.close, onClick: action("On Clear") }}
      dataCy="input-text"
      initialValue="Adornment is clickable and triggers an action"
      label="Label"
    />
  </StoriesWrapper>
);

export const WithPlaceholder = () => (
  <StoriesWrapper>
    <InputText dataCy="input-text" placeholder="Type something..." label="Label" shrink />
  </StoriesWrapper>
);

export const WithIntl = () => (
  // IntlProviderMock simulates external IntlProvider context
  // The `InputTextIntl` exported version of `InputText` uses `labelId` prop to get a localized label.
  // When using `InputTextIntl` the prop `dataCy` can be omitted as it defaults to `labelId` value.
  <StoriesWrapper>
    <IntlProviderMock locale={select("locale", LocaleMock, LocaleMock.en)}>
      <InputTextIntl labelId={MessageMock.inputText} onChange={action("Click on Button")} />
    </IntlProviderMock>
  </StoriesWrapper>
);
