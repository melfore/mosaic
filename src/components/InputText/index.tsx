import React, { FC, useState, useEffect } from 'react';
import MUITextField from '@material-ui/core/TextField';
import { InputSize, InputTextProps, InputVariant, MultilineInputType } from '../../types/InputText';

const getMultilineProps = (multiline?: MultilineInputType) => {
  return {
    multiline: !!multiline,
    ...(!multiline ? {} : { ...multiline }),
  }
};

/**
 * InputText component made on top of `@material-ui/core/TextField`
 */
const InputText: FC<InputTextProps> = ({
  dataCy = 'input-text',
  disabled = false,
  initialValue = '',
  // TODO#lb: implement labelId
  label = 'Label',
  multiline = undefined,
  onChange = undefined,
  required = false,
  size = InputSize.default,
  variant = InputVariant.default,
}) => {
  const [value, setValue] = useState(initialValue);
  useEffect(() => setValue(initialValue), [initialValue]);

  const onChangeHandler = event => {
    const { target: { value }} = event;
    setValue(value);
    if (onChange) {
      onChange(value);
    }
  };

  return (
    <MUITextField
      disabled={disabled}
      inputProps={{
        'data-cy': dataCy,
      }}
      InputProps={{
        readOnly: disabled,
      }}
      label={label}
      margin="normal"
      onChange={onChangeHandler}
      required={required}
      size={size}
      variant={variant}
      value={value}
      {...getMultilineProps(multiline)}
    />
  );
};

export default InputText;
