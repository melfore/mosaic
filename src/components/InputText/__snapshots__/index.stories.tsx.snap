// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs/InputText Adornment smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r1:"
         id=":r1:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-mui-classname">
    <input aria-invalid="false"
           id=":r1:"
           type="text"
           data-cy="input-text"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-mui-classname"
           value="Value"
           style="width: 100%;"
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-mui-classname">
      <button class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
              tabindex="-1"
              type="button"
              disabled
              data-cy="icon-button"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-mui-classname"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="CloseIcon"
             data-cy="icon-button-icon"
        >
          <path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z">
          </path>
        </svg>
      </button>
    </div>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputText AdornmentClickable smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r2:"
         id=":r2:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-mui-classname">
    <input aria-invalid="false"
           id=":r2:"
           type="text"
           data-cy="input-text"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-mui-classname"
           value="Value"
           style="width: 100%;"
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-mui-classname">
      <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
              tabindex="0"
              type="button"
              data-cy="icon-button"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-mui-classname"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="CloseIcon"
             data-cy="icon-button-icon"
        >
          <path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z">
          </path>
        </svg>
      </button>
    </div>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputText Disabled smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r3:"
         id=":r3:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl Mui-readOnly MuiInputBase-readOnly css-mui-classname">
    <input aria-invalid="false"
           disabled
           id=":r3:"
           readonly
           type="text"
           data-cy="input-text"
           class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-readOnly MuiInputBase-readOnly css-mui-classname"
           value="Value"
           style="width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputText Localized smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r4:"
         id=":r4:-label"
  >
    Campo testuale
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":r4:"
           placeholder="Inserire valore..."
           type="text"
           data-cy="locale.inputText"
           class="MuiInputBase-input MuiOutlinedInput-input css-mui-classname"
           value="Value"
           style="width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Campo testuale
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputText Multiline smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r5:"
         id=":r5:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-mui-classname">
    <textarea aria-invalid="false"
              id=":r5:"
              data-cy="input-text"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-mui-classname"
              style="width: 100%; height: 69px;"
    >
      Value
    </textarea>
    <textarea aria-hidden="true"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-mui-classname"
              readonly
              tabindex="-1"
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0px); width: 1220px; padding-top: 0px; padding-bottom: 0px;"
    >
    </textarea>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputText Placeholder smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="false"
         for=":r6:"
         id=":r6:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":r6:"
           placeholder="Enter value..."
           type="text"
           data-cy="input-text"
           class="MuiInputBase-input MuiOutlinedInput-input css-mui-classname"
           value
           style="width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputText Primary play-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r0:"
         id=":r0:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":r0:"
           type="text"
           data-cy="input-text"
           class="MuiInputBase-input MuiOutlinedInput-input css-mui-classname"
           value="ValueChanged"
           style="width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputText Required smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r7:"
         id=":r7:-label"
  >
    Label
    <span aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-mui-classname"
    >
       *
    </span>
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":r7:"
           required
           type="text"
           data-cy="input-text"
           class="MuiInputBase-input MuiOutlinedInput-input css-mui-classname"
           value="Value"
           style="width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label *
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputText SizeSmall smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r8:"
         id=":r8:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-mui-classname">
    <input aria-invalid="false"
           id=":r8:"
           type="text"
           data-cy="input-text"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-mui-classname"
           value="Value"
           style="width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputText Styled smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r9:"
         id=":r9:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":r9:"
           type="text"
           data-cy="input-text"
           class="MuiInputBase-input MuiOutlinedInput-input css-mui-classname"
           value="Value"
           style="width: 100%; color: red; font-weight: bold; font-size: large; text-align: center;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputText VariantFilled smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-filled css-mui-classname"
         data-shrink="true"
         for=":ra:"
         id=":ra:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":ra:"
           type="text"
           data-cy="input-text"
           class="MuiInputBase-input MuiFilledInput-input css-mui-classname"
           value="Value"
           style="width: 100%;"
    >
  </div>
</div>
`;

exports[`Inputs/InputText VariantStandard smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-mui-classname"
         data-shrink="true"
         for=":rb:"
         id=":rb:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":rb:"
           type="text"
           data-cy="input-text"
           class="MuiInputBase-input MuiInput-input css-mui-classname"
           value="Value"
           style="width: 100%;"
    >
  </div>
</div>
`;
